extensions/amp-story-interactive/0.1/amp-story-interactive-slider.js

```diff
@@ -115,17 +119,68 @@ export class AmpStoryInteractiveSlider extends AmpStoryInteractive {
       (event) => event.stopPropagation(),
       true
     );
+
+    this.storeService_.subscribe(
+      StateProperty.CURRENT_PAGE_ID,
+      (currPageId) => {
+        if (currPageId != this.getPageEl().getAttribute('id')) {
+          return;
+        }
+        const DURATION_MS = 1500;
+        let startTime;
+        const animateFrame = (currTime) => {
+          // Set current startTime if not defined.
+          if (!startTime) {
+            startTime = currTime;
+          }
+          const elapsed = currTime - startTime;
+          if (elapsed < DURATION_MS) {
+            // Value between 0 and 1;
+            const timePercentage = elapsed / DURATION_MS;
+            // example with Math.sin animation
+            // Converts oscillation between 1 and -1, to 0 and 30.
+            const val = map(
+              Math.cos(timePercentage * Math.PI * 2),
+              1,
+              -1,
+              0,
+              30
+            );
+            this.inputEl_.value = val;
+            this.onDrag_();
+            if (!this.isInteracting_) {
+              requestAnimationFrame(animateFrame);
+            }
+          }
+        };
+        requestAnimationFrame(animateFrame);
+      }
+    );
+    /**
+     * @private
+     * @param {number} num
+     * @param {number} inMin
+     * @param {number} inMax
+     * @param {number} outMin
+     * @param {number} outMax
+     * @return {!Element}
+     */
+    function map(num, inMin, inMax, outMin, outMax) {
+      return ((num - inMin) * (outMax - outMin)) / (inMax - inMin) + outMin;
+    }
```

Comment:
We can probably find an easier way to map these numbers. Let's find out the function we want to use, and then we can easily use sth like `(num) => num * rangeWeight + rangeOffset`